### PR TITLE
added support for specifying user data for the worker launch config t…

### PR DESCRIPTION
**Pull request by jimbeck**
In reference to #42

### CODE REVIEW

**Comment by brandoconnor** [comment]
Hey @jimbeck -

This looks good to me. The one change I'd request is aligning the userdata variable names. Let's make what you have here conventional and alter the other to be `post_userdata`. There's a spot in the test fixture where this should be altered as well.

Also, be sure to add your change to the CHANGELOG to be released under v1.3.0.

File: data.tf [hunk 1]
```diff
@@ -73,6 +73,7 @@ data template_file userdata {
     endpoint            = "${aws_eks_cluster.this.endpoint}"
     cluster_auth_base64 = "${aws_eks_cluster.this.certificate_authority.0.data}"
     max_pod_count       = "${lookup(local.max_pod_per_node, lookup(var.worker_groups[count.index], "instance_type", lookup(var.workers_group_defaults, "instance_type")))}"
+    pre_userdata        = "${lookup(var.worker_groups[count.index], "pre_userdata",lookup(var.workers_group_defaults, "pre_userdata"))}"
     additional_userdata = "${lookup(var.worker_groups[count.index], "additional_userdata",lookup(var.workers_group_defaults, "additional_userdata"))}"
```
Comment (brandoconnor) [inline data.tf]: Just needing to get this in alignment wherever it appears.

Comment (jimbeck) [reply data.tf]: Will do. Would you prefer that we only change the internal additional_userdata to post_userdata or would you prefer to introduce a breaking change?

Comment (jimbeck) [reply data.tf]: cool I updated the changelog for now reflecting the `pre_userdata` variable so that it's captured.

Comment (brandoconnor) [reply data.tf]: Thank you!

**Pull request merged**